src/txn/txn_recover.c

```diff
@@ -867,6 +874,45 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
     r.files[0].c = NULL;
     WT_ERR(metac->close(metac));
 
+rollback_to_stable:
+    /*
+     * Perform rollback to stable only when the following conditions met.
+     * 1. The connection is not read-only. A read-only connection expects that there shouldn't be
+     *    any changes that need to be done on the database other than reading.
+     * 2. The history store file was found in the metadata.
+     */
+    if (hs_exists && !F_ISSET(conn, WT_CONN_READONLY)) {
+        /* Start the eviction threads for rollback to stable if not already started. */
+        WT_ERR(__wt_evict_create(session));
+        eviction_started = true;
+
```

Comment:
Can we move the `WT_ERR(__recovery_set_checkpoint_snapshot(session));` call to here as it is used only in the rollback to stable code?

Reply:
Yeah, moved the function before calling rts.